wandb/sdk/wandb_run.py

```diff
@@ -3206,27 +3264,46 @@ def link_model(
             run.link_model(
                 path="/local/directory",
                 registered_model_name="my_entity/my_project/my_reg_model",
-                model_name="my_model_artifact",
+                name="my_model_artifact",
+                aliases=["production"],
+            )
+
+            run.link_model(
+                path="/local/directory",
+                registered_model_name="my_reg_model",
+                name="my_entity/my_project/my_model_artifact",
                 aliases=["production"],
             )
             ```
 
         Raises:
-            AssertionError: if registered_model_name is a path
+            AssertionError: if registered_model_name is a path or
+                if model artifact 'name' is of a type that does not contain the substring 'model'
+            ValueError: if name has invalid special characters
 
         Returns:
             None
         """
-        name_parts = linked_model_name.split("/")
+        name_parts = registered_model_name.split("/")
         assert (
             len(name_parts) == 1
         ), "Please provide only the name of the registered model. Do not append the entity or project name."
         project = "model-registry"
-        target_path = self.entity + "/" + project + "/" + linked_model_name
+        target_path = self.entity + "/" + project + "/" + registered_model_name
 
-        artifact = self._log_artifact(
-            artifact_or_path=path, name=model_name, type="model"
-        )
+        public_api = self._public_api()
+        try:
+            artifact = public_api.artifact(name=f"{name}:latest")
+            assert "model" in str(
+                artifact.type.lower()
+            ), "You can only use this method for 'model' artifacts. For an artifact to be a 'model' artifact, its type property must contain the substring 'model'."
```

Comment:
same as above

Reply:
Nope! This assertion is to make sure users only link artifacts of type 'model', similar to how we handle it in `use_model()`